src/producer/browser/Partial.js

```diff
@@ -24,15 +24,36 @@ import onSplitsArrivedFactory from './onSplitsArrivedFactory';
 const PartialBrowserProducer = (context) => {
   const settings = context.get(context.constants.SETTINGS);
   const { splits: splitsEventEmitter } = context.get(context.constants.READINESS);
-  
+
   const segmentsUpdater = MySegmentsUpdater(context);
   const segmentsUpdaterTask = TaskFactory(segmentsUpdater, settings.scheduler.segmentsRefreshRate);
 
   const onSplitsArrived = onSplitsArrivedFactory(segmentsUpdaterTask, context);
-  
+
   splitsEventEmitter.on(splitsEventEmitter.SDK_SPLITS_ARRIVED, onSplitsArrived);
 
-  return segmentsUpdaterTask;
+  let isSynchronizeMySegmentsRunning = false;
+
+  function synchronizeMySegments(segmentList) {
```

Comment:
Shouldn't you assign this to the task so you can keep track of those updates too? Or should you remove the splits one from the full producer splits task?

Reply:
done